mina.netty/src/main/java/org/jboss/netty/channel/socket/nio/AbstractNioWorker.java

```diff
@@ -618,4 +643,53 @@ public void run() {
             }
         });
     }
+
+    public void register(NioChildDatagramChannel childChannel) {    // TODO no datagram specific
```

Comment:
One possibility would be to have the channel indicate if it needs to be registered with a selector, and default to `true` so that most the code is not impacted, then have the child datagram channel indicate `false`, causing selector registration to be skipped only when necessary.